### ISSUE TITLE
Register Type years

### ISSUE THREAD

**Issue by edickens** [FreeUKGen/MyopicVicar#1437]

**Comment by edickens** [comment]
Registers for active churches are still be entered.  So the Register Type years for these will not have an End year.  So I propose that "Today" can be entered in the End year.
This will cause the validation of uploaded batch entries and Image Source years to have all dates validated against today, or an end year if that is entered because the Church is no longer active.  It also means that the End year does not need to keep being changed.  I think  this is why many Register Type years have not been entered.

**Comment by PatReynolds** [comment]
I am not sure that "Register type years" is the right terminology - it is the last year which a transcriber is permitted to transcribe (which rolls on with time).

**Comment by Sherlock21** [comment]
I agree Pat, that the title is a bit of a mouthful but what would be simpler and precise?

I also agree with Eric D's idea as the opening comment. A nice easy & tidy solution.

**Comment by Captainkirkdawson** [comment]
In the original specification. The coordinator was to enter a start and end date and the system would use that to validate the date entered in the transcription.

**Comment by SteveBiggs** [comment]
@Captainkirkdawson that is still what should happen, but if there is no end date (because records are still being entered in that register) could the end date be stated as "today" and roll on with time?

**Comment by Captainkirkdawson** [comment]
Are we talking about the same fields?
![years](https://user-images.githubusercontent.com/2874694/39190777-8d155346-4792-11e8-8e9c-6f10d7f422f6.PNG)
The first 2 actual years are computed from the data set
The permitted years are those for the register. 
If those dates are absent we use a system defined minimum and maximum year. The minimum year is 1300. The maximum year is today's year.
Hence I fail to understand the requirement.

**Comment by SteveBiggs** [comment]
I wasn't aware that the system already assumes the current year for the Last permitted year if nothing is entered. In that case it sounds like no change is needed. Presumably one can enter just a First permitted year and leave the Last year blank so it defaults to current year?

**Comment by edickens** [comment]
It is the permitted years, those which are used for validation, that we are talking about.
In this case probably starting in 1597, but if the church is still entering into their registers, then the Last permitted year is Today.  So do we either leave it blank or enter "Today"?  If blank, then we have a start but no end entry.  I usually like something positive so that we know it was deliberately not entered.
